dynamodb-lambda-elasticache/README.md

```diff
@@ -0,0 +1,78 @@
+# CQRS pattern with Amazon DynamoDB, AWS Lambda and Amazon Elasticache for Redis
+
+This project contains an event-driven microservices design pattern that is influenced by the [CQRS Pattern](https://microservices.io/patterns/data/cqrs.html).
```

Comment:
im just dont know how this pattern makes it faster than a normal read to DynamoDB. Have you tested that? DynamoDB has a single milisecond performance and doesnt need to be behind a VPC to work. It would be great to get this pattern backed by some data before creating it

Reply:
This pattern focuses on microservices architecture use-cases where each microservice owns one or more sub-domains and a database dedicated to the domain. Due to this architecture pattern, it's no longer starightforward to implement queries and join data from multiple services. For example, for ordering service, each service is responsible for different writes like create-order service. And to "get order order details" we might need to join from multiple tables like order, payment, customer details etc. Since both reading and writing have different approaches, we can employ different strategies for each operation. CQRS (Command and Query Responsibility Segregation) uses “separation of concern” principle to separate the read database from the write database. 

Commands perform mutation operations such as “create order” or “update order”. Queries perform ready-only operations such as “Get order details”. By this isolation, we can design purpose-built database based on the usage requirement. With the CQRS pattern, we could implement multiple denormalised views that make the application scalable and performant by simplifying the command query models because of its separation of concern principle.

In this pattern we are using DynamoDB for the high-performant writes and Elasticache for building materialised views for the application reads